core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,45 +16,52 @@
  * @fileoverview Service to send changes to a story to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/story/story-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('EditableStoryBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_STORY_DATA_URL_TEMPLATE', 'STORY_PUBLISH_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService,
-      EDITABLE_STORY_DATA_URL_TEMPLATE, STORY_PUBLISH_URL_TEMPLATE) {
-    var _fetchStory = function(storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import { StoryDomainConstants } from
+  'domain/story/story-domain.constants.ts';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableStoryBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
 
-      $http.get(storyDataUrl).then(function(response) {
-        var story = angular.copy(response.data.story);
-        var topicName = angular.copy(response.data.topic_name);
-        var storyIsPublished = response.data.story_is_published;
-        var skillSummaries = angular.copy(response.data.skill_summaries);
+  private storyDataDict = null;
+  private _fetchStory(
+      storyName: string, successCallback: any, errorCallback: any): any {
+    var editableStoryDataUrl = this.urlInterpolation.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_name: storyName
+      });
+
+    this.http.get(
+      editableStoryDataUrl, { observe: 'response' }).toPromise().then(
+      (response: any) => {
+        this.storyDataDict = cloneDeep(response.body);
         if (successCallback) {
-          successCallback({
-            story: story,
-            topicName: topicName,
-            storyIsPublished: storyIsPublished,
-            skillSummaries: skillSummaries
-          });
+          successCallback(this.storyDataDict);
         }
-      }, function(errorResponse) {
+      }, (errorResponse: any) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.body);
```

Comment:
Use "error" property for error responses. Change here and other places in this file.
eg. `errorCallback(errorResponse.error);`

Reply:
Done.